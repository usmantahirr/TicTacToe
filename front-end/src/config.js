export const API_URL = "http://localhost:3001/api/";
export const TIMEOUT = "1000";
export const COLORS = {
  PRIMARY: '#2c3e50',
  NEUTRAL: '#ecf0f1',
  DANGER: '#e74c3c',
  BORDER: '#34495e',
  BORDER_LIGHT: '#bdc3c7',
  HOVER: '#ecf0f1',
  FONT_LIGHT: 'white',
  FONT_DARK: '#2c3e50',
};
