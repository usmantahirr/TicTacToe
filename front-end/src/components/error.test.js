import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import { COLORS } from "../config";
import Error from "./error";

describe("Error", () => {
  it("Should render passed prop correctly", () => {
    const component = shallow(<Error message="Error Message" />);

    expect(component).toMatchSnapshot();
    expect(component.text()).toEqual("Error Message");
  });

  it("Should have correct styles", () => {
    const tree = renderer.create(<Error message="Error Message" />).toJSON();
    expect(tree).toHaveStyleRule("background", COLORS.DANGER);
    expect(tree).toHaveStyleRule("color", COLORS.FONT_LIGHT);
  });
});
