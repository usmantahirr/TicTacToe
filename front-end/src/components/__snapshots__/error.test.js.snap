// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Should render passed prop correctly 1`] = `
<styled.div>
  Error Message
</styled.div>
`;
